compiler/src/dotty/tools/dotc/typer/Dynamic.scala

```diff
@@ -176,10 +178,30 @@ trait Dynamic {
    *  It's an error if U is neither a value nor a method type, or a dependent method
    *  type
    */
-  def handleStructural(tree: Tree)(using Context): Tree = {
+  def handleStructural(tree: Tree)(using Context) = {
     val fun @ Select(qual, name) = funPart(tree): @unchecked
     val vargss = termArgss(tree)
 
+    def handleRepeated(base: untpd.Tree, possiblyCurried: List[List[Tree]], isReflectSelectable: Boolean) = {
+      val handled = possiblyCurried.map { args =>
+        val isRepeated = args.exists(_.tpe.widen.isRepeatedParam)
+        if isRepeated && isReflectSelectable then
+          List(untpd.TypedSplice(tpd.repeatedSeq(args, TypeTree(defn.AnyType))))
+        else args.map { t =>
+          val clzSym = t.tpe.resultType.classSymbol.asClass
+          if ValueClasses.isDerivedValueClass(clzSym) && isReflectSelectable then
+            val underlying = ValueClasses.valueClassUnbox(clzSym).asTerm
+            tpd.Select(t, underlying.name)
+          else
+            t
+        }.map(untpd.TypedSplice(_))
+      }
+
+      if isReflectSelectable
+      then untpd.Apply(base, handled.flatten)
+      else handled.foldLeft(base)((base, args) => untpd.Apply(base, args))
```

Comment:
```suggestion
      untpd.Apply(base, handled.flatten)
```
We should always flatten the argument lists for `Selectable`

Reply:
thanks! I will fix that